src/ctype/CType.ts

```diff
@@ -10,40 +10,57 @@
  * @preferred
  */
 
+import { validateAddress } from '../util/DataUtils'
 import { CTypeWrapperModel } from './CTypeSchema'
 import CTypeUtils from './CType.utils'
 import ICType, { CompressedCType } from '../types/CType'
 import Identity from '../identity/Identity'
-import { getOwner, store } from './CType.chain'
+import { store, getOwner } from './CType.chain'
 import TxStatus from '../blockchain/TxStatus'
 import IClaim from '../types/Claim'
 
 export default class CType implements ICType {
   public static fromCType(cTypeInput: ICType): CType {
-    if (!CTypeUtils.verifySchema(cTypeInput, CTypeWrapperModel)) {
+    return new CType(cTypeInput)
+  }
+
+  public static fromSchema(
+    schema: ICType['schema'],
+    owner?: ICType['owner']
+  ): CType {
+    return new CType({
+      hash: CTypeUtils.getHashForSchema(schema),
+      owner: owner || null,
+      schema,
+    })
+  }
+
+  static isICType(input: ICType): input is ICType {
+    if (!CTypeUtils.verifySchema(input, CTypeWrapperModel)) {
       throw new Error('CType does not correspond to schema')
     }
-    if (cTypeInput.hash) {
-      if (CTypeUtils.getHashForSchema(cTypeInput.schema) !== cTypeInput.hash) {
-        throw Error('provided and generated cType hash are not matching')
-      }
+    if (CTypeUtils.getHashForSchema(input.schema) !== input.hash) {
```

Comment:
This is for deserialization purposes. So if we did not check this and just initialize it, we might have an initialized CType with a hash not corresponding to it's schema.
Our current approach is that the constructor should do as little as possible in calculating the instanciated objects property values, so it has to take a whole ICType as input, with hash already calculated and checked against the schema.

Reply:
Altho this does smells not very dry, kinda redundant, but let's see.
Maybe we'll change where things have to be checked, and where it would be redundant.

Reply:
An alternative would be to check schema & owner on the constructor and store a calculated hash, and only allow passing a hash on `fromCtype()` where we then compare it to the hash on a Ctype we instantiated using only schema & owner.